### DISCUSSION TITLE
Setting up Nightscout - Install Nightscout phase 2 -  keep getting errors. Retry 10+ times

### DISCUSSION THREAD

**Discussion by smtuhkanen** [NightscoutFoundation/xDrip#3369] [Bug reports]
I cant get the nightscout phase 2 to install. I keep getting the error. I followed the instructions when errors are encountered. Waited a couple mins and tried over and over. Closed the window and started over.
 Maybe I am doing something wrong? What is my API_SECRET? do i make that up or is it from one of the previous steps (i have been tackling this over many weeks.)

**Comment by Navid200** [comment]
When you run phase 2 for the very first time, you choose an API_SECRET.  
If you press escape, it will set it to the default value.

When you run it again, if you press escape, it keeps whatever API_SECRET was before.

In either case, when it asks you to choose an API_SECRET, it shows you the existing value for it.

You say you keep getting an error.  Can you take a screenshot of the error and post here so that I can see?

**Reply by smtuhkanen** [reply]
After entering API Secret and my login and password. This prompt shows up. I choose 2 per the instructions.
![image](https://github.com/NightscoutFoundation/xDrip/assets/160206296/32fd9eb5-610a-4631-8688-57100afbb860)

Then I get this message. I choose OK
![image](https://github.com/NightscoutFoundation/xDrip/assets/160206296/20ccf0b7-afa9-44c7-b664-89990ae15d20)

Then it immediately gives this. And then I repeat over again. I have also closed the SSH and reopened and started from the beginning. 
![image](https://github.com/NightscoutFoundation/xDrip/assets/160206296/38090aa4-f091-43b4-add1-a79bb39bf9ce)

**Reply by Navid200** [reply]
So, you have done everything correctly.
What is the error?

Have you looked at the video clips?

**Reply by smtuhkanen** [reply]
Yeah I followed the clips throughout the set up. All I get is connection failed. I was thinking maybe I should go back and try a different site than chickenkiller...

**Reply by Navid200** [reply]
What does the status page look like?
What happens if you use a browser to access Nightscout?

**Reply by smtuhkanen** [reply]
I am using SSH-in-browser

![image](https://github.com/NightscoutFoundation/xDrip/assets/160206296/0e7d7058-3900-4e95-87e1-3355e76cce8f)

**Reply by Navid200** [reply]
What happens if you open a browser and enter your hostname as the address?

**Reply by smtuhkanen** [reply]
Site cant be reached

**Reply by Navid200** [reply]
That's what I wanted to see a screenshot of.  That's what I meant by the error.

**Reply by Navid200** [reply]
Please delete the image.  You have disclosed your hostname.
There is nothing wrong with it.
When I entered it, I got the prompt to authenticate.

Don't you get that?

**Reply by smtuhkanen** [reply]
no - but maybe it is because i am working off a company computer?

**Reply by smtuhkanen** [reply]
ok I tried off my phone - and I get to an authentication page... but the dozen + times I was trying phase 2 I had been trying different API Secrets.. is there a way that I can reset or find it out? Or am I screwed?

**Reply by Navid200** [reply]
Please ask the question differently.  I don't understand your question.

**Reply by Navid200** [reply]
OK, may be I understand.
Your API_SECRET is one of the Nightscout variables.
You can attempt to edit them just to see them without editing.  Then, you can see your API_SECRET.
You can also see your API secret on the status page at the bottom by selecting login credentials.

And this explain how to edit variables:  https://navid200.github.io/xDrip/docs/Nightscout/NS_Variables.html

**Reply by smtuhkanen** [reply]
Sorry, I was able to remember the API Secret. It appears to be working as intended. Thanks for your help!

**Reply by Navid200** [reply]
You're welcome
